server/test_helpers.go

```diff
@@ -13,16 +13,10 @@ func FreeTCPAddr() (addr, port string, err error) {
 		return "", "", err
 	}
 
-	closer := func() {
-		err := l.Close()
-		if err != nil {
-			// TODO: Handle with #870
-			panic(err)
-		}
+	if err := l.Close(); err != nil {
+		return "", "", fmt.Errorf("couldn't close the listener: %w", err)
```

Comment:
should we wrap it?
```
sdkerrors.Wrap(sdkerrors.ErrIO, ....)
```

Reply:
Yes, sure, good one. I'm not just yet sure this PR actually fixes anything at all. I warmly welcome suggestions - and I'm CC'ing @odeke-em too.

If this does not improve anything, I'd just close it.

Reply:
The PR has good improvement - it removes `panic`

Reply:
[thinking aloud]
shouldn't I call Close() as the very last statement of this function?
[/thinking aloud]

Reply:
Yup, @robert-zaremba got a great point that we should wrap it, for consistency with other errors.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/8879